pkg/types/interfacetests/chain_reader_interface_tests.go

```diff
@@ -213,6 +215,228 @@ func runChainReaderGetLatestValueInterfaceTests[T TestingT[T]](t T, tester Chain
 	runTests(t, tester, tests)
 }
 
+func runChainReaderBatchGetLatestValueInterfaceTests[T TestingT[T]](t T, tester ChainReaderInterfaceTester[T]) {
+	testCases := []testcase[T]{
+		{
+			name: "BatchGetLatestValue works",
+			test: func(t T) {
+				// setup test data
+				firstItem := CreateTestStruct(1, tester)
+				batchCallEntry := make(BatchCallEntry)
+				batchCallEntry[AnyContractName] = ContractBatchEntry{{Name: MethodTakingLatestParamsReturningTestStruct, ReturnValue: &firstItem}}
+				tester.SetBatchLatestValues(t, batchCallEntry)
+
+				// setup call data
+				params, actual := &LatestParams{I: 1}, &TestStruct{}
+				batchGetLatestValueRequest := make(types.BatchGetLatestValueRequest)
+				batchGetLatestValueRequest[AnyContractName] = []types.BatchRead{{ReadName: MethodTakingLatestParamsReturningTestStruct, Params: params, ReturnVal: actual}}
+
+				ctx := tests.Context(t)
+				cr := tester.GetChainReader(t)
+
+				require.NoError(t, cr.Bind(ctx, tester.GetBindings(t)))
+				result, err := cr.BatchGetLatestValue(ctx, batchGetLatestValueRequest)
+				require.NoError(t, err)
+
+				anyContractBatch := result[AnyContractName]
+				assert.Equal(t, MethodTakingLatestParamsReturningTestStruct, anyContractBatch[0].ReadName)
+				assert.Equal(t, &firstItem, anyContractBatch[0].ReturnValue)
+				assert.NoError(t, anyContractBatch[0].Err)
+			},
+		},
+		{
+			name: "BatchGetLatestValue works without arguments and with primitive return",
+			test: func(t T) {
+				// setup call data
+				var primitiveReturnValue uint64
+				batchGetLatestValueRequest := make(types.BatchGetLatestValueRequest)
+				batchGetLatestValueRequest[AnyContractName] = []types.BatchRead{{ReadName: MethodReturningUint64, Params: nil, ReturnVal: &primitiveReturnValue}}
+
+				ctx := tests.Context(t)
+				cr := tester.GetChainReader(t)
+				require.NoError(t, cr.Bind(ctx, tester.GetBindings(t)))
+
+				result, err := cr.BatchGetLatestValue(ctx, batchGetLatestValueRequest)
+				require.NoError(t, err)
+
+				anyContractBatch := result[AnyContractName]
+				require.NoError(t, anyContractBatch[0].Err)
+				assert.Equal(t, MethodReturningUint64, anyContractBatch[0].ReadName)
+				assert.Equal(t, AnyValueToReadWithoutAnArgument, *anyContractBatch[0].ReturnValue.(*uint64))
+			},
+		},
+		{
+			name: "BatchGetLatestValue allows a contract name to resolve different contracts internally",
```

Comment:
does `TestContract` resolve to different contracts?

Reply:
hmm, good catch, I copied some tests from GetLatestValue, the title is a bit misleading

Reply:
Actually my bad, this test does resolve different contracts in chain impl. tests